src/app.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react'
+import FlashCardForm from './flashcardform'
+
+export default class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      cardInfo: JSON.parse(localStorage.getItem('cardInfo')) || [],
+      lastId: JSON.parse(localStorage.getItem('lastId')) || 0
+    }
+    this.handleSave = this.handleSave.bind(this)
+  }
+  componentDidMount() {
+    window.addEventListener('beforeunload', () => {
+      const {cardInfo, lastId} = this.state
+      const newCardInfo = cardInfo.map((newCard) => {
```

Comment:
You don't need to copy your cards before stringifying them. `JSON.stringify` does not mutate its input.